### PR TITLE
mempool:  Add a skeleton mempool pkg

### PR DESCRIPTION
**Pull request by dajohi**
[Review on Reviewable](https://reviewable.io/reviews/btcsuite/btcd/571)

### CODE REVIEW

**Comment by davecgh** [comment]
I'd prefer to not create the skeleton package unless it's absolutely needed.  The mining case was required because mempool separation would be blocked.  In this case, there is nothing else that needs `mempoolPolicy`.  In other words, I'd just introduce `type mempoolPolicy struct` and switch the code to make use of it (very close to what you've done here, but without the package).

**Comment by dajohi** [comment]
How does this look?

File: config.go [hunk 1]
```diff
@@ -337,7 +337,7 @@ func loadConfig() (*config, []string, error) {
 		BlockMaxSize:      defaultBlockMaxSize,
 		BlockPrioritySize: defaultBlockPrioritySize,
 		SigCacheMaxSize:   defaultSigCacheMaxSize,
-		MaxOrphanTxs:      maxOrphanTransactions,
+		MaxOrphanTxs:      1000,
```
Comment (davecgh) [inline config.go]: This should be a `defaultMaxOrphanTransactions` constant to be consistent with the surrounding code.

File: mempool.go [hunk 2]
```diff
@@ -64,21 +46,12 @@ type mempoolConfig struct {
...
-	MaxOrphanTxs int
-
-	// MinRelayTxFee defines the minimum transaction fee in BTC/kB to be
-	// considered a non-zero fee.
-	MinRelayTxFee btcutil.Amount
-
 	// NewestSha defines the function to retrieve the newest sha
 	NewestSha func() (*wire.ShaHash, int32, error)
 
+	// Policy
```
Comment (davecgh) [inline mempool.go]: The comment is incomplete.

File: mempool.go [hunk 3]
```diff
@@ -1103,9 +1110,10 @@ func (mp *txMemPool) LastUpdated() time.Time {
 
 // newTxMemPool returns a new memory pool for validating and storing standalone
 // transactions until they are mined into a block.
-func newTxMemPool(cfg *mempoolConfig) *txMemPool {
+func newTxMemPool(policy *mempoolPolicy, cfg *mempoolConfig) *txMemPool {
```
Comment (davecgh) [inline mempool.go]: The `mempoolConfig` struct has the `Policy` field in it.  There shouldn't be a need to pass them both.

File: mempool.go [hunk 4]
```diff
@@ -1103,9 +1110,10 @@ func (mp *txMemPool) LastUpdated() time.Time {
...
 	memPool := &txMemPool{
 		cfg:           *cfg,
+		policy:        *policy,
```
Comment (davecgh) [inline mempool.go]: `cfg.Policy`

File: server.go [hunk 5]
```diff
@@ -2362,19 +2362,24 @@ func newServer(listenAddrs []string, db database.Db, chainParams *chaincfg.Param
...
 		EnableAddrIndex:       cfg.AddrIndex,
 		FetchTransactionStore: s.blockManager.blockChain.FetchTransactionStore,
-		FreeTxRelayLimit:      cfg.FreeTxRelayLimit,
-		MaxOrphanTxs:          cfg.MaxOrphanTxs,
-		MinRelayTxFee:         cfg.minRelayTxFee,
 		NewestSha:             s.db.NewestSha,
 		RelayNtfnChan:         s.relayNtfnChan,
 		SigCache:              s.sigCache,
 		TimeSource:            s.timeSource,
 	}
```
Comment (davecgh) [inline server.go]:
```Go
	Policy: mempoolPolicy{
		...
	},
```

**Comment by davecgh** [comment]
This looks good overall.  In addition to addressing the line comments, this needs a rebase since it can't be merged cleanly.

**Comment by dajohi** [comment]
Rebased.

**Comment by davecgh** [comment]
Thanks for addressing the review items.  I've reviewed all of the updates and tested them.

OK

**Pull request merged**